Add customizable number of points in simple runner.

krokosik opened the pull request:
## Description

My use case has a function that is already fast due to parallel execution. As such, not only does it evaluate quicker than 50ms, but also freezes my PC when run with parallel runner. As such, I found the `simple` runner very useful, but it also introduces overhead by running the interpolants on ever new point. Increasing the `ask` count to 16 (my number of cores to mimic the parallel version), diminished that to overhead to almost nothing, but I would prefer to not have to monkey patch this function :)

## Checklist

- [x] Fixed style issues using `pre-commit run --all` (first install using `pip install pre-commit`)
- [x] `pytest` passed

## Type of change

*Check relevant option(s).*

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] (Code) style fix or documentation update
- [ ] This change requires a documentation update

Commit:
Add customizable number of points for simple runner ask

[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Commit:
Fix minor typing issue in learner 2D

akhmerov commented on adaptive/runner.py:
```diff
@@ -911,11 +911,12 @@ def simple(
     npoints_goal: int | None = None,
     end_time_goal: datetime | None = None,
     duration_goal: timedelta | int | float | None = None,
+    points_per_ask: int | None = 1,
```
```suggestion
    points_per_ask: int = 1,
```

krokosik replied:
This would make the type checker yell at the end user for not providing this arguments.

akhmerov reviewed (changes requested):
This is nice and simple, looks helpful too! Please add a test to confirm that `points_per_ask > 1` works, and I think this PR is good to go!

krokosik commented:
While I understand the need for full test coverage, this is a bit problematic here. The `simple` runner is basically a blocking while loop and there is no way to check the `pending_points` after `ask` has been called.

I guess we could mock this method, but I'm not that experienced with `pytest`, I guess it would involve adding the `pytest-mock` plugin, but do we want to go through that for such a simple functionality that does not introduce any changes for existing code?

basnijholt committed:
Add test for points_per_ask parameter in simple runner

Test the new points_per_ask parameter added in PR #484 which allows
customizing the number of points requested between interpolation reruns.
This reduces overhead for fast functions by batching point requests.

The test verifies:
- Parameter works with 1D and 2D learners
- Default behavior (points_per_ask=1) is preserved
- Performance improvement via reduced ask() calls with higher values

basnijholt committed:
use `points_per_ask: int = 1`

basnijholt reviewed (approved):
Made the change @akhmerov suggested (which I fully agree with). @krokosik I don't see how this is a problem with type checkers?

Also added a test.
